### PR TITLE
SW-4710 Add cohorts table, add cohorts_phase table

### PR DESCRIPTION
**Pull request by nickgraz**
- Add a new table to hold cohort data
- Add a new enum table to hold cohort phase enums
- Add relationship between cohort and cohort phase within cohort table
- Add the phases from the PRD as enums into the `cohort_phases` table

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1677** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1677?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #1678** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1678?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1679** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1680** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1680?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1681** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1681?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1677?utm_source=stack-comment).

File: src/main/resources/db/migration/0200/V243__Cohorts.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+-- Add cohort phase enum table (phase 0, phase 1, etc...)
+CREATE TABLE cohort_phases (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL,
+    UNIQUE (name)
+);
+
+-- Add cohort / phase relationship
+ALTER TABLE cohorts ADD COLUMN phase_id INTEGER REFERENCES cohort_phases;
```
Comment (sgrimm) [inline src/main/resources/db/migration/0200/V243__Cohorts.sql]: A couple comments on this:

1. If you create `cohort_phases` first, then this column can be part of the `CREATE TABLE` statement.
2. Can a cohort ever not be in a phase? I suspect this should be a `NOT NULL` column.

File: jooq/src/main/kotlin/com/terraformation/backend/jooq/Config.kt [hunk 2]
```diff
@@ -174,6 +175,8 @@ val ID_WRAPPERS =
             listOf(
                 IdWrapper("AutomationId", listOf("automations\\.id")),
                 IdWrapper("BalenaDeviceId", listOf("device_managers\\.balena_id")),
+                IdWrapper("CohortId", listOf("cohorts\\.id", ".*\\.cohort_id")),
+                IdWrapper("CohortPhaseId", listOf("cohort_phases\\.id", ".*\\.cohort_phase_id")),
```
Comment (sgrimm) [inline jooq/src/main/kotlin/com/terraformation/backend/jooq/Config.kt]: If the table is turned into an enum, you shouldn't need an ID wrapper class. I'm kind of surprised having both at the same time doesn't break the build!

**Review by sgrimm — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/0200/V243__Cohorts.sql (hunk 1 above)
Comment (nickgraz) [reply src/main/resources/db/migration/0200/V243__Cohorts.sql]: Ah, you're right, cohort must be created with a phase as it is required

**Review by sgrimm — APPROVED** [review]

**Pull request merged**